arrow-ipc/src/writer.rs

```diff
@@ -547,6 +564,57 @@ impl IpcDataGenerator {
     }
 }
 
+fn set_variadic_buffer_counts(counts: &mut Vec<i64>, array: &dyn Array) {
```

Comment:
Maybe a name like 

```suggestion
fn append_variadic_buffer_counts(counts: &mut Vec<i64>, array: &dyn Array) {
```

Would be clearer?

I think it would also help to clarify that this code adds variadic buffers for nested types as well